spec/rspec/matchers/built_in/raise_error_spec.rb

```diff
@@ -1,11 +1,38 @@
 RSpec.describe "expect { ... }.to raise_error" do
   it_behaves_like("an RSpec matcher", :valid_value => lambda { raise "boom" },
                                       :invalid_value => lambda { }) do
-    let(:matcher) { raise_error }
+    let(:matcher) { raise_error Exception }
   end
 
   it "passes if anything is raised" do
-    expect {raise}.to raise_error
+    expect { raise "error" }.to raise_error "error"
+  end
+
+  it "issues a warning when used without an error class or message" do
+    expect_warning_with_call_site __FILE__, __LINE__+1, /without providing a specific error/
+    expect { raise }.to raise_error
+  end
+
+  it "can supresses the warning when configured to do so" do
+    original_value = RSpec::Expectations.configuration.warn_about_false_positives?
+    begin
+      RSpec::Expectations.configuration.warn_about_false_positives = false
+      expect { raise }.to raise_error
+    ensure
+      RSpec::Expectations.configuration.warn_about_false_positives = original_value
+    end
+  end
```

Comment:
Should the config toggling be moved into a shared context that handles that detail, like these?

https://github.com/rspec/rspec-expectations/blob/a75fe7bfc48788c724e51fb6e1b4581b76376e19/spec/spec_helper.rb#L53-L99